src/script/components/ModalComponent.tsx

```diff
@@ -88,6 +88,23 @@ const ModalComponent: React.FC<ModalComponentProps> = ({
   const [displayNone, setDisplayNone] = useState<boolean>(!isShown);
   const hasVisibleClass = isShown && !displayNone;
   const isMounting = useRef<boolean>(true);
+  const trapId = useId();
+
+  const onKeyDown = (event: KeyboardEvent): void => {
```

Comment:
I think we should use a `useCallback` here, since we want the reference to the function to be stable over re-renders.

Reply:
Also please rename the `onKeyDown` to a more imperative style function name (like `trapFocus`) that would explain more what the function is actually doing. 

I know we have a history of lazy naming event handler as the name of the event they are reacting to, but I think it's a bad practice that completely hides what the purpose of the function is